ETWAnalyzer_uTest/EventDump/DumpFileTests.cs

```diff
@@ -37,9 +36,63 @@ public void DumpFileCheck()
                 FileRenameCount = 3,
                 FileOpenCloseTimeInus = 201,
             };
+
+            dump.SortOrder = ETWAnalyzer.Commands.DumpCommand.SortOrders.OpenCloseTime;
             decimal totalTime = dump.GetSortValue(groupedData);
             Assert.Equal(201, totalTime);
 
+            dump.SortOrder = ETWAnalyzer.Commands.DumpCommand.SortOrders.Length;
+            decimal length = dump.GetSortValue(groupedData);
+            Assert.Equal(20, length);
+
+            dump.SortOrder = ETWAnalyzer.Commands.DumpCommand.SortOrders.Size;
+            decimal size = dump.GetSortValue(groupedData);
+            Assert.Equal(212, size);
+
+            dump.SortOrder = ETWAnalyzer.Commands.DumpCommand.SortOrders.Time;
+            decimal time = dump.GetSortValue(groupedData);
+            Assert.Equal(212, time);
+
+            dump.SortOrder = ETWAnalyzer.Commands.DumpCommand.SortOrders.Count;
+            decimal count = dump.GetSortValue(groupedData);
+            Assert.Equal(212, size);
+
+            MatchData groupedData01 = new()
```

Comment:
Why did you define a second groupdata? I think the second test would be to test SortOrder and a value for FileOperation which can be Read,Write,Open,Close,SetSecurity,Delete,Rename which filters away all other files which do not have at least one such operation defined. 

Please define as test values something which can be easily calculated by a human and not some arbitrary values like 64, 256. It is sufficient to use nice values like 1, 5, 10, 100, 500, 1000 which are easy to differentiate and add up easily.

Reply:
completed